ui.js

```diff
@@ -0,0 +1,128 @@
+// Set up event listeners for submit & stop buttons
+const submitButton = document.getElementById('submit')
+const stopButton = document.getElementById('stop')
+
+
+var resetSignal = () => {
+	abortController = new AbortController()
+	globalAbortSignal = abortController.signal
+}
+
+submitButton.addEventListener('click', evt => {
+	// Clear any ongoing requests, then load page
+	abortController.abort()
+	// create new signals
+	resetSignal()
+
+	loadPage()
+})
+
+stopButton.addEventListener('click', async (evt) => {
+	// Clear any ongoing requests, then stop loading of images
+	abortController.abort()
+
+	// create new signals
+	resetSignal()
+	stopItAll()
+})
+
+// by UndertowTruck
+// sorts all images displayed (related functions in misc.js)
+sortOptions.addEventListener('change', evt => {
```

Comment:
I think there needs to be a button to actually do the sorting, instead of relying on state changes. When I first used it, it wasn't clear to me that in order to make things sort, I need to change the selection in the dropdown. A button to do it would be clearer.

As an example of why this is better - I have a ton of favorites for my user id, and it can take 10+ min to load them all. As it's loading, I like to sort by score. Then, I want to sort again after it's loaded more images. Without a button, the way to do this is to change the dropdown to something else, then change it back to "score". It would be much better to just have a button to click.

Reply:
Good idea. I wanted the UI to become instantly reactive whenever you made a change, but you raise a good point. I'll also add an apply filters button for the filters section as well